share/functions/__fish_cached.fish

```diff
@@ -0,0 +1,35 @@
+function __fish_cached --description "Cache the command output for a given amount of time"
+
+    argparse --min-args 1 --stop-nonopt 't/max-age=!_validate_int --min 0' 'k/cache-key=' -- $argv
+    or return
+
+    if set -q _flag_cache_key
+        set -f cache_key $_flag_cache_key
+    else
+        set -f cache_key (string split ' ' "$argv" | head -1)
```

Comment:
Note: an idiomatic way

	set -f cache_key (string split ' ' "$argv")[1]

because that avoids the need for a fork / new process.
No big deal though, since this is not called often.